endpoint.go

```diff
@@ -944,6 +944,10 @@ func (ep *endpoint) assignAddressVersion(ipVer int, ipam ipamapi.Ipam) error {
 			ep.Unlock()
 			return nil
 		}
+		if err == ipamapi.ErrNoIPReturned && ipam.capability.SupportsAutoIPv6 {
```

Comment:
I do not think this function needs this change.
If the IPAM driver, say, does not handle IPv4, then I would expect `len(ipInfo) == 0` above, and this function can exit, regardless of `ipVer` (need to change this).

Because I would expect `ipamAllocateVersion()` to not initialize `network.ipamV4Info` if the IPAM driver does not or does not want to support autoallocation of IPv4 pools.